pkg/pillar/types/global.go

```diff
@@ -103,312 +109,439 @@ func (gs *GlobalStatus) UpdateItemValuesFromGlobalConfig(gc GlobalConfig) {
+	if _, ok := configPtr.GlobalSettings[key]; ok {
+		configPtr.GlobalSettings[key] = ConfigItemValue{stringValue: newValue}
+	} else {
+		return fmt.Errorf("unable to modify global setting %s because setting was not found", string(key))
 	}
-	if newgc.VdiskGCTime < GlobalConfigMinimums.VdiskGCTime {
-		log.Warnf("Enforce minimum VdiskGCTime received %d; using %d",
-			newgc.VdiskGCTime, GlobalConfigMinimums.VdiskGCTime)
-		newgc.VdiskGCTime = GlobalConfigMinimums.VdiskGCTime
+	return nil
+}
+
+func (configPtr *ConfigItemValueMap) DelAgentValue(key AgentSettingKey, agentName string) error {
+	// I don't think this is needed because if you are calling delete, the key must already exist. But maybe just in case the user makes a mistake?
+	if configPtr.AgentSettings == nil {
+		configPtr.AgentSettings = make(map[string]map[AgentSettingKey]ConfigItemValue)
+	}
+	if settingMap, ok := configPtr.AgentSettings[agentName]; ok {
+		if _, okVal := settingMap[key]; okVal {
+			delete(settingMap, key)
+		} else {
+			return fmt.Errorf("unable to delete per-agent setting for agent: %s because setting %s was not found",
+				agentName, string(key))
+		}
+	} else {
+		return fmt.Errorf("unable to delete per-agent setting for agent: %s because agent was not found",
+			agentName)
+	}
+	return nil
+}
+
+func (configPtr *ConfigItemValueMap) ResetGlobalValue(key GlobalSettingKey) error {
+	specMap := NewConfigItemSpecMap()
+	if _, ok := configPtr.GlobalSettings[key]; ok {
+		configPtr.GlobalSettings[key] = specMap.GlobalSettings[key].ParseSpecToValue()
+	} else {
+		return fmt.Errorf("unable to modify global setting %s because setting was not found", string(key))
 	}
-	if newgc.DownloadRetryTime < GlobalConfigMinimums.DownloadRetryTime {
-		log.Warnf("Enforce minimum DownloadRetryTime received %d; using %d",
-			newgc.DownloadRetryTime, GlobalConfigMinimums.DownloadRetryTime)
-		newgc.DownloadRetryTime = GlobalConfigMinimums.DownloadRetryTime
-	}
-	if newgc.DomainBootRetryTime < GlobalConfigMinimums.DomainBootRetryTime {
-		log.Warnf("Enforce minimum DomainBootRetryTime received %d; using %d",
-			newgc.DomainBootRetryTime, GlobalConfigMinimums.DomainBootRetryTime)
-		newgc.DomainBootRetryTime = GlobalConfigMinimums.DomainBootRetryTime
-	}
-	if newgc.RktGCGracePeriod < GlobalConfigMinimums.RktGCGracePeriod {
-		log.Warnf("Enforce minimum RktGCGracePeriod received %d; using %d",
-			newgc.RktGCGracePeriod, GlobalConfigMinimums.RktGCGracePeriod)
-		newgc.RktGCGracePeriod = GlobalConfigMinimums.RktGCGracePeriod
-	}
-	if newgc.Dom0MinDiskUsagePercent < GlobalConfigMinimums.Dom0MinDiskUsagePercent {
-		log.Warnf("Enforce minimum Dom0MinDiskUsagePercent received %d; using %d",
-			newgc.Dom0MinDiskUsagePercent, GlobalConfigMinimums.Dom0MinDiskUsagePercent)
-		newgc.Dom0MinDiskUsagePercent = GlobalConfigMinimums.Dom0MinDiskUsagePercent
-	}
-	return newgc
+	return nil
 }
+
+func (configPtr *ConfigItemValueMap) FormatConfig() ConfigItemValueMap {
```

Comment:
You are Essentially resetting configPtr to its default state.. Is it? Is that the purpose of this routine? Should we just have:
    DefaultConfigItemValueMap() function which just returns a default Map?

Curious -- why do you need this routine?